### PR TITLE
tests: split travis spread execution in 2 jobs for ubuntu and non ubuntu systems

### PR DESCRIPTION
**Pull request by sergiocazzolato**
It also includes to run together the main suite with the upgrade suite
to improve the jobs time and avoid lack of jobs.

In case of lack of jobs on travis, we can add more vms to reduce the execution time.

### CODE REVIEW

File: run-checks [hunk 1]
```diff
@@ -55,6 +55,12 @@ case "${1:-all}" in
     --spread)
         SPREAD=1
```
Comment (mvo5) [inline run-checks]: Would it make sense to have `SPREAD=full` here?

File: run-checks [hunk 2]
```diff
@@ -55,6 +55,12 @@ case "${1:-all}" in
...
         ;;
+    --spread-ubuntu)
+        SPREAD=2
```
Comment (mvo5) [inline run-checks]: And SPREAD=ubuntu-only here?

File: run-checks [hunk 3]
```diff
@@ -55,6 +55,12 @@ case "${1:-all}" in
...
+        ;;
+    --spread-no-ubuntu)
+        SPREAD=3
```
Comment (mvo5) [inline run-checks]: And SPREAD=no-ubuntu here? This way the checks below will convey a bit more context.

File: run-checks (hunk 1 above)
Comment (zyga) [reply run-checks]: I agree with mvo. There's no reason to use integers for this.

**Comment by mvo5** [comment]
There is something funny going on - from the log:
```
./run-checks: 267: [: Illegal number: ubuntu-only
```
whats strange about this is that the script does not exit with an error :/

File: run-checks [hunk 4]
```diff
@@ -258,14 +264,23 @@ if [ "$UNIT" = 1 ]; then
...
-if [ "$SPREAD" = 1 ]; then
+if [ "$SPREAD" -gt 0 ]; then
     TMP_SPREAD="$(mktemp -d)"
     addtrap "rm -rf \"$TMP_SPREAD\""
 
     export PATH=$TMP_SPREAD:$PATH
     ( cd "$TMP_SPREAD" && curl -s -O https://niemeyer.s3.amazonaws.com/spread-amd64.tar.gz && tar xzvf spread-amd64.tar.gz )
 
-    spread google: google-upgrade:tests/upgrade/
+    if [ "$SPREAD" = full ]; then
```
Comment (zyga) [inline run-checks]: Perhaps this should be a switch statement?

Comment (sergiocazzolato) [reply run-checks]: Yes, thanks for the review!!

File: spread.yaml [hunk 5]
```diff
@@ -687,7 +657,6 @@ suites:
         systems: [-ubuntu-core-*, -opensuse-*]
         # autopkgtest runs against localhost which causes problems with
```
Comment (mvo5) [inline spread.yaml]: I think we also need to remove this comment now - autopkgtest does no longer run the "upgrade" test.

Comment (sergiocazzolato) [reply spread.yaml]: done

**Pull request closed without merging**